openassessment/workflow/models.py

```diff
@@ -364,11 +420,27 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
+            sub_api.set_score(
+                self.submission_uuid,
+                score["points_earned"],
+                score["points_possible"]
+            )
+
+    def staff_score_exists(self):
+        """
+        Check if a staff score exists for this submission. Cached after first use.
```

Comment:
How is this getting cached? I'm happy to hear it is, but I'm not sure what the mechanism is for it.